paymentsheet/src/main/java/com/stripe/android/paymentsheet/ui/HeaderTextFactory.kt

```diff
@@ -16,55 +16,69 @@ internal class HeaderTextFactory(
         types: List<PaymentMethodCode>,
     ): Int? {
         return if (isCompleteFlow) {
-            when (screen) {
-                is PaymentSheetScreen.SelectSavedPaymentMethods -> {
-                    if (isWalletEnabled) {
-                        null
-                    } else {
-                        R.string.stripe_paymentsheet_select_payment_method
-                    }
-                }
-                is PaymentSheetScreen.AddFirstPaymentMethod, PaymentSheetScreen.VerticalMode -> {
-                    R.string.stripe_paymentsheet_add_payment_method_title.takeUnless {
-                        isWalletEnabled
-                    }
-                }
-                is PaymentSheetScreen.EditPaymentMethod -> {
-                    StripeR.string.stripe_title_update_card
-                }
-                is PaymentSheetScreen.Loading,
-                is PaymentSheetScreen.AddAnotherPaymentMethod,
-                is PaymentSheetScreen.Form,
-                null -> {
-                    null
-                }
-            }
+            createForCompleteFlow(screen, isWalletEnabled)
         } else {
-            when (screen) {
-                is PaymentSheetScreen.Loading, is PaymentSheetScreen.Form -> {
-                    null
-                }
-                is PaymentSheetScreen.SelectSavedPaymentMethods -> {
-                    R.string.stripe_paymentsheet_select_payment_method
-                }
-                is PaymentSheetScreen.AddFirstPaymentMethod,
-                is PaymentSheetScreen.AddAnotherPaymentMethod,
-                is PaymentSheetScreen.VerticalMode -> {
-                    val title = if (types.singleOrNull() == PaymentMethod.Type.Card.code) {
-                        StripeR.string.stripe_title_add_a_card
-                    } else {
-                        R.string.stripe_paymentsheet_choose_payment_method
-                    }
+            createForFlowController(screen, types, isWalletEnabled)
+        }
+    }
+
+    private fun createForCompleteFlow(
+        screen: PaymentSheetScreen?,
+        isWalletEnabled: Boolean
+    ) = when (screen) {
+        is PaymentSheetScreen.SelectSavedPaymentMethods -> {
+            if (isWalletEnabled) {
+                null
+            } else {
+                R.string.stripe_paymentsheet_select_payment_method
+            }
+        }
+        is PaymentSheetScreen.AddFirstPaymentMethod, PaymentSheetScreen.VerticalMode -> {
+            R.string.stripe_paymentsheet_add_payment_method_title.takeUnless {
+                isWalletEnabled
+            }
+        }
+        is PaymentSheetScreen.EditPaymentMethod -> {
+            StripeR.string.stripe_title_update_card
+        }
+        is PaymentSheetScreen.ManageSavedPaymentMethods -> {
+            R.string.stripe_paymentsheet_select_payment_method
```

Comment:
Can you add a test for this (if it's the final variant).

I've been skipping tests for all the stuff I've added, since it isn't done yet.